### PR TITLE
[refactor] use posix for Process.groups

### PR DESCRIPTION
**Pull request by kares**
instead of relying on *com.sun.security.auth.module* internals

this change drops `org.jruby.platform.Platfom.getPlatform` and its (private) sub-classes,
`Process.groups` on Windows is still "not implemented" as before

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyProcess.java [hunk 1]
```diff
@@ -605,12 +605,16 @@ public static IRubyObject exit_bang(IRubyObject recv, IRubyObject[] args) {
 
     @JRubyMethod(name = "groups", module = true, visibility = PRIVATE)
     public static IRubyObject groups(IRubyObject recv) {
-        long[] groups = Platform.getPlatform().getGroups(recv);
-        RubyArray ary = RubyArray.newArray(recv.getRuntime(), groups.length);
+        final Ruby runtime = recv.getRuntime();
+        long[] groups = runtime.getPosix().getgroups();
+        if (groups == null) { // not-implemented for the given platform (e.g. Windows)
+            throw runtime.newNotImplementedError("groups() function is not implemented");
```
Comment (ahorek) [inline core/src/main/java/org/jruby/RubyProcess.java:611]:
```suggestion
            throw runtime.newNotImplementedError("groups() function is unimplemented on this machine");
```
\* to match MRI's message

**Review by ahorek — APPROVED** [review]
nice cleanup

**Comment by headius** [comment]
This cleanup seems fine, but it's not zero-sum. On systems where the user has chosen to disable native support, or if native support can't be loaded for some other reason (security etc) there's no fallback to the old logic.

I would suggest that perhaps the invasive reflection-based logic be added to JavaPosix in jnr-posix so we can still support this functionality without native support.

**Comment by headius** [comment]
Status here @kares? Can we restore the old logic as a fallback and get this merged in for 9.3?

**Comment by kares** [comment]
sorry haven't had the time to do this as the jnr-posix end, yet

**Comment by headius** [comment]
@kares Now's the time to merge or punt to 9.4.1.0.

**Comment by kares** [comment]
believe, it's a good idea to merge and not rely on JDK internals, despite not having the jnr-posix work in place.
the only regression I can see potentially happening from this would be `Process.groups` not working on Solaris (haven't checked if it's actually the case).

**Pull request merged**